src/pysdmx/io/xml/sdmx21/reader/data_read.py

```diff
@@ -0,0 +1,342 @@
+"""Module that holds the necessary functions to read xml files."""
+
+import itertools
+from typing import Any, Dict, List, Optional
+
+import numpy as np
+import pandas as pd
+
+from pysdmx.io.xml.sdmx21.__parsing_config import (
+    AGENCY_ID,
+    ATTRIBUTES,
+    DIM_OBS,
+    exc_attributes,
+    GROUP,
+    ID,
+    OBS,
+    OBS_DIM,
+    OBSKEY,
+    OBSVALUE,
+    REF,
+    SERIES,
+    SERIESKEY,
+    STR_USAGE,
+    STRID,
+    STRREF,
+    STRSPE,
+    STRUCTURE,
+    URN,
+    VALUE,
+    VERSION,
+)
+from pysdmx.util.handlers import add_list, split_from_urn
+
+chunksize = 50000
+
+
+class Dataset:
```

Comment:
I guess that, eventually, we'll want to create a Dataset class under pysdmx.model, isn't it? And, like all other model classes, it should be format-agnostic? If so, I think we can go ahead with this class now but we should discuss about the dataset class and create it before the 1.0.0 release. What do you think?

Reply:
Sorry, forget about the above. I see you already mentioned the same in your PR :)

Reply:
Yes, closing this as it will be modified in #49 and some discussions are pending.